repositories/mentor_repository.js

```diff
@@ -19,34 +20,80 @@ async function getMentorByUserId(userId) {
 
 async function getMentorById(allowedStatuses, selectedFields, mentorId) {
   const knex = getKnex();
-  return await knex("mentors")
+  const mentor = await knex("mentors")
     .join("users", "mentors.user_id", "users.id")
     .where("mentors.id", mentorId)
     .whereIn("mentors.status", allowedStatuses)
     .select(selectedFields).first();
+
+  const mentorTags = await tagsRepo.getAssignedTagNames(
```

Comment:
same, I would suggest removing all tagsRepo calls from this repository and rather move them onto the service layer